[pm-5966] Fix Entity Framework query for MySQL

JimmyVo16 opened the pull request:
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-5966

## 📔 Objective

Problem: The Entity Framework query was causing a compile-time error.

Changes:

1. Fixed the query.
2. Renamed the variable to replace the comment.

## 📸 Screenshots

Manual testing 

Generated SQL query.
![image](https://github.com/user-attachments/assets/ce2710ee-d439-4eb9-b6c7-47ae9bcda106)

With results

<img width="754" alt="image" src="https://github.com/user-attachments/assets/18a0e1de-99b7-4afa-921b-c03dabf2034f" />



## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

<!-- Suggested interactions but feel free to use (or not) as you desire! -->

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

JimmyVo16 committed:
[pm-5966] Fix Entity Framework query for MySQL

JimmyVo16 commented on src/Infrastructure.EntityFramework/Repositories/OrganizationDomainRepository.cs:
```diff
@@ -147,14 +147,13 @@ from s in sJoin.DefaultIfEmpty()
-            .Where(x => (DateTime.UtcNow - x.CreationDate).Days == 4
-                        && x.VerifiedDate == null)
+        var unverifiedDomainsInLast72Hours = await dbContext.OrganizationDomains
+            .Where(x => x.CreationDate.Date >= DateTime.UtcNow.AddDays(-4).Date
```
Question: I'm not sure if a unit test would add value here since the behavior can only be tested at runtime.

r-tome replied:
You can add an integration test on the project `Infrastructure.IntegrationTest` using the `DatabaseData` attribute which configures the test to run once for each configured DB provider.
Example here https://github.com/bitwarden/server/blob/main/test/Infrastructure.IntegrationTest/AdminConsole/Repositories/OrganizationRepositoryTests.cs

r-tome reviewed (changes requested):
It looks good to me but it would be ideal to have the integration test

holow29 commented:
Looks like this should resolve #3722.  Funny to see it resolved because of a compilation issue vs. the pre-existing issue.

JimmyVo16 committed:
Add integration tests.

JimmyVo16 commented on test/Infrastructure.IntegrationTest/AdminConsole/Repositories/OrganizationDomainRepositoryTests.cs:
```diff
@@ -0,0 +1,136 @@
+public class OrganizationDomainRepositoryTests
+{
+    [DatabaseTheory, DatabaseData]
+    public async Task GetExpiredOrganizationDomainsAsync_ShouldReturnExpiredOrganizationDomains(
```
hey @eliykat 

I’m trying to add a test for the VerifiedDate field, and I need a way to update it. My plan is to create a class that extends OrganizationDomain and add test logic to update the VerifiedDate, since this will only be used in test env. 

Just wanted to check if this is an acceptable pattern here or if there’s a better way to handle this.

r-tome replied:
Which specific test are you trying to add? It looks like you're already using the `.SetVerifiedDate()` method here, and it seems to be fulfilling the purpose for these tests.

r-tome replied:
Maybe you meant `CreationDate`? Well, that one seems tricky. Maybe you can use `ITimeProvider` to set the value and then mock it on the test?

JimmyVo16 replied:
Sorry, I meant `CreationDate`. Thanks, I'll look into `ITimeProvider`.

JimmyVo16 replied:
After a nice weekend of rest, this is actually easier than I thought, lol. I added the new tests.

JimmyVo16 committed:
Merge branch 'main' into ac/pm-5966/fix-entity-framework-query

JimmyVo16 committed:
Add tests for CreationDate.

JimmyVo16 committed:
Add assertion comment.

JimmyVo16 committed:
Merge branch 'main' into ac/pm-5966/fix-entity-framework-query

JimmyVo16 committed:
Fix failed store proc tests

- Replace incorrect models to simplify compatibility with the stored proc.

- Fix logic issues and update tests to ensure consistency between stored proc and Entity Framework.
